docs/Conversion/GPUToSPIRV.md

```diff
@@ -0,0 +1,152 @@
+# GPUXToSPIRVPass
```

Comment:
The recommended way of providing docs is in .td files. There is a whole machinery to extract and assemble/group docs from various places to generate a consistent doc structure. Individual doc files counteract this approach.

Reply:
I have added description in .td file as well, but because I added some examples for this pass extension and that is a bit lengthy, that would have clobbered .td file with unnecessary details, so thought of adding them in separate .md file.

Reply:
Look at upstream passes, most of them (not all, though) have all the docs in the .td file, no matter how long they are (like https://github.com/llvm/llvm-project/blob/129420df5122a27560a17f6a472c8c750d3bbd7b/mlir/include/mlir/Dialect/Bufferization/Transforms/Passes.td).

Spreading docs across multiple places makes maintenance even worse. Less files, less policies, less trouble.

What's the point of having an extra file?

Reply:
I didn't see one of these .td file which included example input and output so assumed td file documentation does not include that level of detail. Thanks for sharing above, I have removed this extra doc file and added examples in .td file.